### PR TITLE
Add notebooks to website

### LINKED ISSUE #142: [Proposal] Add example notebooks as pages to documentation
It looks like it is possible to add Jupyter notebooks as part of documentation with the following extension: https://pypi.org/project/mkdocs-jupyter/

### PR DESCRIPTION
**Pull request by SemyonSinchenko**
Close #142 

Add all notebooks as pages to the website. That's how it looks like:
image

1. I moved notebooks from `examples` to `docs/examples`
2. I added `poetry` group `mkdocs`
3. I added `mkdocs-jupyter` dependency
4. I modified `website.yml` to use poetry (to have dependencies consistency)

Changes to be committed:
	modified:   .github/workflows/website.yml
	new file:   docs/examples/create_udf.ipynb
	renamed:    examples/example.ipynb -> docs/examples/example.ipynb
	renamed:    examples/spark_ai_cache.json -> docs/examples/spark_ai_cache.json
	renamed:    examples/transform_dataframe.ipynb -> docs/examples/transform_dataframe.ipynb
	renamed:    examples/vector_similarity_search.ipynb -> docs/examples/vector_similarity_search.ipynb
	deleted:    examples/create_udf.ipynb
	modified:   mkdocs.yml
	modified:   poetry.lock
	modified:   pyproject.toml

### CODE REVIEW

**Comment by SemyonSinchenko** [comment]
I have no idea why tests were failed because I did not touch anything except adding new poetry group and modifying mkdocs

**Comment by SemyonSinchenko** [comment]
@gengliangwang Hello!

This PR is ready to review and checks are green.

**What was changed** (I marked as bold important changes):
- **added an mkdocs plugin for notebooks**;
- **site generation was switched to poetry as a single source of truth for all the dependencies; to achieve that the 'website' dependency group was introduced**;
- **example notebooks were moved to 'docs', otherwise it is not working**;
- updated website ci (added trigger 'by dispatch' to allow one to see how the site looks like before merging to main);
- updated dependencies;
- for some notebooks output was missed, I updated them;

I build the site from the source branch in my fork, you may check how it looks: https://semyonsinchenko.github.io/pyspark-ai/

So, please, make a look and give your feedback! Thanks in advance!

File: mkdocs.yml
Comment (gengliangwang) [inline mkdocs.yml:24]: Shall we move this one to the first so that users can have a quick overview?

File: docs/notebooks/create_udf.ipynb
Comment (gengliangwang) [inline docs/notebooks/create_udf.ipynb:1]: why this file is not a clean file rename?

**Comment by gengliangwang** [comment]
@SemyonSinchenko Thanks for the work! I will merge and deploy the changes after comments are addressed.

File: mkdocs.yml
Comment (SemyonSinchenko) [reply mkdocs.yml:23]: Done

**Comment by SemyonSinchenko** [comment]
@gengliangwang The last commit:
- reorder examples in `mkdocs.yml` based on your comment

**Review by gengliangwang — APPROVED** [review]

**Comment by gengliangwang** [comment]
@SemyonSinchenko Thanks for enhancing the doc site!

**Pull request merged**
